### PR TITLE
TAO-7277 - Add service that handles iframe protection

### PR DESCRIPTION
**Pull request by martijn-tao**
~- [ ] Depends on changes in https://github.com/oat-sa/generis/pull/562~

### CODE REVIEW

File: actions/class.CommonModule.php [hunk 1]
```diff
@@ -49,14 +50,16 @@
...
-    protected $service = null;
+    protected $service;
 
     /**
      * tao_actions_CommonModule constructor.
      */
     public function __construct()
     {
-        $this->setIframeHeaders();
+        /** @var ActionProtector $actionProtectionService */
```
Comment (siwane) [inline actions/class.CommonModule.php]: Could u avoid to use the constructor please, as the DI is not already set (ServiceManager)

File: actions/class.Main.php [hunk 2]
```diff
@@ -333,7 +333,7 @@ public function index()
             //check if the user is a noob, otherwise redirect him to his last visited extension.
             $firstTime = TaoCe::isFirstTimeInTao();
             if ($firstTime == false) {
-               $lastVisited = TaoCe::getLastVisitedUrl();
+               $lastVisited = null; //TaoCe::getLastVisitedUrl();
```
Comment (siwane) [inline actions/class.Main.php]: What?

**Comment by siwane** [comment]
Don't forget to bump version

File: actions/class.Main.php [hunk 3]
```diff
@@ -381,35 +388,40 @@ public function index()
...
+        ];
         $this->setData('client_config_url', $this->getClientConfigUrl($clientConfigParams));
-        $this->setData('content-template', array('blocks/sections.tpl', 'tao'));
+        $this->setData('content-template', ['blocks/sections.tpl', 'tao']);
 
-		$this->setView('layout.tpl', 'tao');
-	}
+        $this->setView('layout.tpl', 'tao');
+
+        /** @var ActionProtector $actionProtector */
```
Comment (martijn-tao) [inline actions/class.Main.php]: These 3 lines were added to set the headers. The other changes are simply inspection errors that have been resolved.

**Pull request merged**